feat (accordion block): added anchor links

kaeizen opened the pull request:
fixes #2888

kaeizen committed:
add setting for anchor

bfintal commented on src/block-components/advanced/edit.js:
```diff
@@ -52,6 +58,19 @@ export const Edit = () => {
 					title={ __( 'General', i18n ) }
 					id="general"
 				>
+					{ allowCustomAnchor && (
```
Ideally, we should only add new things to block components if they are usable by other blocks, but in this case, this option is specific only to the Accordion block. Structurally, this option should not be present here.

That being said, since this option is just an Anchor to the Accordion block, we do not need to create a new option for this because WP already added one under **Advanced Tab > Advanced Panel** as the `anchor` attribute. Let's use that attribute and option instead.

bfintal reviewed (changes requested):
Some comments when I tried out the new feature:
1. When opening/closing the accordion with an anchor, it shouldn't scroll the page
2. When I opened the url with a hash in another window, the accordion didn't open
3. When I opened the url with a hash in another window, the page didn't scroll to the accordion

kaeizen committed:
update fix

- use WP Advanced Panel anchor attribute & option
- prevent scrolling when opening/closing accordion
- open and scroll to accordion on page load of URL with hash

kaeizen committed:
removed unused variables

kaeizen committed:
remove unused import
